build.gradle

```diff
@@ -107,6 +107,7 @@ dependencies {
     implementation "org.opensearch:opensearch:${opensearch_version}"
     implementation 'org.junit.jupiter:junit-jupiter:5.10.0'
     implementation "com.google.guava:guava:32.1.2-jre"
+    implementation 'javax.ws.rs:javax.ws.rs-api:2.1.1'
```

Comment:
I think this dependency is only here for the `Response.Status` used for the exception, but you can use `org.opensearch.core.rest.RestStatus` for that and remove this dependency.
```suggestion
```

Reply:
Removed this dependency earlier but it stayed in my local, didn't get a chance to push it until I was offcall :)